### ISSUE TITLE
[java] UseStringBufferForStringAppends: Wrong example in documentation

### ISSUE THREAD

**Issue by Belle-PL** [pmd/pmd#3374]
**Affects PMD Version:** 

**Rule:UseStringBufferForStringAppends**

Please provide the rule name and a link to the rule documentation:
<https://pmd.github.io/latest/pmd_rules_java_performance.html#usestringbufferforstringappends>

**Description:**
Hi, I have encountered a false negative when PMD tried to detect **UseStringBufferForStringAppends**. Below is the test case for reproducing this false negative:

**Code Sample demonstrating the issue:**

```java
public void testUseStringBufferForStringAppends() {
    String a;
    a = "foo";
    a += " bar"; // Should report a warning here
}
```

**Expected outcome:Warning**

PMD should report a violation at line 4, but doesn't. This is a false-negative.

**Running PMD through:** *[Maven]*

**Comment by Belle-PL** [comment]
@adangel Hi Andreas, could you please help me check this issue? Thanks!

**Comment by adangel** [comment]
Hi @Belle-PL ,

the example we have on our documentation is too trivial. Exactly this case has been removed from detection with #1736 . If you create manually a StringBuilder to concatenate two strings, it will be slower - the java compiler optimizes this already for you. See also the comment at #1736.

Btw. the example is of course so trivial, that you would actually just write `String a = "foo bar"` instead of concatenating constant strings.

We probably should use a more complex example, like

```java
String result = "";
for (int i = 0; i < 10; i++) {
    result += getStringFromSomeWhere(i);
}
```

This indeed should be written using a StringBuilder.

**Comment by Belle-PL** [comment]
Thanks for your kind response! Hope this case will be removed soon. I think it is a little misleading.
